R/condition_call_linter.R

```diff
@@ -0,0 +1,98 @@
+#' Recommend usage of `call. = FALSE` in conditions
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "stop('test')",
+#'   linters = condition_call_linter()
+#' )
+#'
+#' lint(
+#'   text = "stop('test', call. = TRUE)",
+#'   linters = condition_call_linter()
+#' )
+#'
+#' lint(
+#'   text = "stop('test', call. = FALSE)",
+#'   linters = condition_call_linter(display_call = TRUE)
+#' )
+#'
+#' lint(
+#'   text = "stop('this is a', 'test', call. = FALSE)",
+#'   linters = condition_call_linter(display_call = TRUE)
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "stop('test', call. = FALSE)",
+#'   linters = condition_call_linter()
+#' )
+#'
+#' lint(
+#'   text = "stop('this is a', 'test', call. = FALSE)",
+#'   linters = condition_call_linter()
+#' )
+#'
+#' lint(
+#'   text = "stop('test', call. = TRUE)",
+#'   linters = condition_call_linter(display_call = TRUE)
+#' )
+#'
+#' @evalRd rd_tags("condition_call_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
```

Comment:
Add a reference link to design.tidyverse.org?

Reply:
I missed the location of this comment. The link is now provided here: https://github.com/Bisaloo/lintr/blob/16a9b3f021bf960fca19e91568a0cecc75f2764e/R/condition_call_linter.R#L3-L5. Please let me know if you don't think it's appropriate.

Reply:
We usually put it in `@seealso`.